server/src/db.ts

```diff
@@ -549,7 +549,15 @@ export async function getCurrentAvailabilityByLocation(
 }
 
 /**
- * Updates a given location's availability based upon its id
+ * Updates a given location's availability. If either `checked_at` or `valid_at`
+ * are not the same or newer than an existing availibility record for the given
+ * location and source, this will throw `OutOfDateError`.
```

Comment:
No, since the conditions on the query may cause 0 rows to update (`checked_at <` vs. `checked_at <=`). It’s a little tricky, I guess — we were doing this same trick before, so it hasn’t become more tricky in *that* respect, except now there are more branches involved which just makes the whole thing harder to track.

Reply:
Oh, I was going to ask next about the `<` vs `<=`, which is incredibly subtle.

Maybe doesn't make sense to block merging on convincing me all is well here. But if you feel inclined, adding tests would probably be more helpful than clarifying documentation or changing the code itself.

Reply:
The tests already cover it pretty well, at least I think so. I did that subtle part wrong at first, and got 4 failures I had to fix. ;)

Reply:
OK, I took your advice on this docstring, and rewrote the bit below to hopefully read more clearly.

I also rewrote the logic so it should be a little more clear and dropped an optimization, which means there’s no clever switching of `<` and `<=` depending on the situation (this is probably the right trade-off). That said, this still has to add a condition on `valid_at` in one branch, but I don’t think I can clean that up without adding locks (which would be good, but I want to get #463 done first).

Reply:
I also realized while re-writing it that `isChange` was named backwards, which probably did not help in terms of this code being understandable. 😬 (It was still functionally doing the right thing; yay tests)